internal/config/moduleconfig/module_config.go

```diff
@@ -214,3 +249,22 @@ func SummarizeConditions(module ModuleConfig) []projectconfig.Condition {
 	}
 	return moduleConditions
 }
+
+// UnmarshalYAML Parses a version constraint string into go-version constraint during yaml parsing
+func (semVer *VersionConstraints) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var versionString string
+	err := unmarshal(&versionString)
+	if err != nil {
+		return err
+	}
+
+	constraints, constErr := goVerson.NewConstraint(versionString)
```

Comment:
If the module.yml doesn't have a version specified we should probably just skip the check.